server/data/data-api/src/main/java/org/apache/james/rrt/lib/Mapping.java

```diff
@@ -57,21 +57,23 @@ static Type detectType(String input) {
     Optional<MailAddress> asMailAddress();
 
     enum Type {
-        Regex("regex:", 4, IdentityMappingBehaviour.Throw),
-        Domain("domain:", 1, IdentityMappingBehaviour.Throw),
-        Error("error:", 4, IdentityMappingBehaviour.Throw),
-        Forward("forward:", 3, IdentityMappingBehaviour.ReturnIdentity),
-        Group("group:", 2, IdentityMappingBehaviour.Throw),
-        Address("", 4, IdentityMappingBehaviour.Throw);
+        Regex("regex:", 4, IdentityMappingBehaviour.Throw, new UserRewritter.RegexRewriter()),
```

Comment:
there's no reuse of Rewriter (one type == on rewriter) so I would implement that as an abstract method of the enum instead.